### PR TITLE
Future proof LPIPs metric

### LINKED ISSUE #1546: LPIPS generates warning.
### To Reproduce

Steps to reproduce the behavior...

```
>>> from torchmetrics.image.lpip import LearnedPerceptualImagePatchSimilarity
>>> lpips = LearnedPerceptualImagePatchSimilarity()
/miniconda/envs/iris/lib/python3.8/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/miniconda/envs/iris/lib/python3.8/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=AlexNet_Weights.IMAGENET1K_V1`. You can also use `weights=AlexNet_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
```

### Expected behavior

No Warning is generated.

### Environment

```
>>> torchvision.__version__
'0.14.1+cu116'
>>> torchmetrics.__version__
'0.11.1'
```

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #1546 
Currently the LPIPs metric relies on the third party package https://github.com/richzhang/PerceptualSimilarity which is not maintained anymore. With the changes to torchvision on how models are loaded, this model will break in the future if we keep relying on the third party package (currently a warning is raised).
This PR copies in the relevant code and updates it to be compatible with torchvision 0.13 and higher.

I have included the required copyright for copying the code.

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/functional/image/lpips.py [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+# limitations under the License.
+
+# Content copied from
+# https://github.com/richzhang/PerceptualSimilarity/blob/master/lpips/lpips.py
+# and
+# https://github.com/richzhang/PerceptualSimilarity/blob/master/lpips/pretrained_networks.py
+# and with adjustments from
+# https://github.com/richzhang/PerceptualSimilarity/pull/114/files
+# Copyright (c) 2018, Richard Zhang, Phillip Isola, Alexei A. Efros, Eli Shechtman, Oliver Wang
+# All rights reserved.
```
Comment (lantiga) [inline src/torchmetrics/functional/image/lpips.py:22]: Let's explicitly 
- clarify the license the source repo has (BSD 2-clause) https://github.com/richzhang/PerceptualSimilarity/blob/master/LICENSE
- clarify that the copy happened because the package was unmaintained

File: src/torchmetrics/functional/image/lpips.py [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+            self.slice4.add_module(str(x), pretrained_features[x])
+        for x in range(10, 11):
+            self.slice5.add_module(str(x), pretrained_features[x])
+        for x in range(11, 12):
+            self.slice6.add_module(str(x), pretrained_features[x])
+        for x in range(12, 13):
+            self.slice7.add_module(str(x), pretrained_features[x])
+        if not requires_grad:
+            for param in self.parameters():
+                param.requires_grad = False
```
Comment (stancld) [inline src/torchmetrics/functional/image/lpips.py:87]:
```suggestion
        self.N_slices = 7
        slices = [torch.nn.Sequential()] * self.N_slices
        for x in range(2):
            slices[0].add_module(str(x), pretrained_features[x])
        for x in range(2, 5):
            slices[1].add_module(str(x), pretrained_features[x])
        for x in range(5, 8):
            self.slices[2].add_module(str(x), pretrained_features[x])
        for x in range(8, 10):
            slices[3].add_module(str(x), pretrained_features[x])
        for x in range(10, 11):
            slices[4].add_module(str(x), pretrained_features[x])
        for x in range(11, 12):
            slices[5].add_module(str(x), pretrained_features[x])
        for x in range(12, 13):
            slices[6].add_module(str(x), pretrained_features[x])
        self.slices = nn.ModuleList(slices)
        if not requires_grad:
            for param in self.parameters():
                param.requires_grad = False
```

File: src/torchmetrics/functional/image/lpips.py [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+        h = self.slice5(h)
+        h_relu5 = h
+        h = self.slice6(h)
+        h_relu6 = h
+        h = self.slice7(h)
+        h_relu7 = h
+        vgg_outputs = namedtuple("SqueezeOutputs", ["relu1", "relu2", "relu3", "relu4", "relu5", "relu6", "relu7"])
+        out = vgg_outputs(h_relu1, h_relu2, h_relu3, h_relu4, h_relu5, h_relu6, h_relu7)
+
+        return out
```
Comment (stancld) [inline src/torchmetrics/functional/image/lpips.py:108]:
```suggestion

    def forward(self, x):
        """Process input."""
        vgg_outputs = namedtuple("SqueezeOutputs", ["relu1", "relu2", "relu3", "relu4", "relu5", "relu6", "relu7"])
        
        relus = []
        for slice in self.slices:
            x = slice(x)
            relus.append(x)
        return vgg_outputs(*relus)
```

And I'd define vgg_outputs `namedtuple` outside of the method/class.

**Review by stancld — COMMENTED** [review]
Some suggestions to the copied code.

**Review by Borda — APPROVED** [review]
